docs/source/high-level-graphs.rst

```diff
@@ -0,0 +1,193 @@
+    ('filter', 0): (lambda part: part[part.name == 'Alice'], ('add', 0)),
+    ('filter', 1): (lambda part: part[part.name == 'Alice'], ('add', 1)),
+    ('filter', 2): (lambda part: part[part.name == 'Alice'], ('add', 2)),
+    ('filter', 3): (lambda part: part[part.name == 'Alice'], ('add', 3)),
+   }
+
+The task graph is a dictionary that stores every Pandas-level function call
+necessary to compute the final result.  We can see that there is some structure
+to this dictionary if we separate out the tasks that were associated to each
+high-level Dask DataFrame operation.
+
+.. code-block:: python
+
+   {
+    # From the dask.dataframe.read_csv call
+    ('read-csv', 0): (pandas.read_csv, 'myfile.0.csv'),
+    ('read-csv', 1): (pandas.read_csv, 'myfile.1.csv'),
+    ('read-csv', 2): (pandas.read_csv, 'myfile.2.csv'),
+    ('read-csv', 3): (pandas.read_csv, 'myfile.3.csv'),
+
+    # From the df + 100 call
+    ('add', 0): (operator.add, 'myfile.0.csv', 100),
+    ('add', 1): (operator.add, 'myfile.1.csv', 100),
+    ('add', 2): (operator.add, 'myfile.2.csv', 100),
+    ('add', 3): (operator.add, 'myfile.3.csv', 100),
+
+    # From the df[df.name == 'Alice'] call
+    ('filter', 0): (lambda part: part[part.name == 'Alice'], ('add', 0)),
+    ('filter', 1): (lambda part: part[part.name == 'Alice'], ('add', 1)),
+    ('filter', 2): (lambda part: part[part.name == 'Alice'], ('add', 2)),
+    ('filter', 3): (lambda part: part[part.name == 'Alice'], ('add', 3)),
+   }
+
+By understanding this high-level structure we are able to understand our task
+graphs more easily (this is more important for larger datasets when there are
+thousands of tasks per layer) and do perform high-level optimizations.  For
+example in the case above we may want to automatically rewrite our code to
+filter our datasets before adding 100
+
+.. code-block:: python
+
+    # Before
+    df = dd.read_csv('myfile.*.csv')
+    df = df + 100
+    df = df[df.name == 'Alice']
+
+    # After
+    df = dd.read_csv('myfile.*.csv')
+    df = df[df.name == 'Alice']
+    df = df + 100
+
+Dask's high level graphs help us to explicitly encode this structure by storing
+our task graphs in layers, with dependencies between layers.
+
+.. code-block:: python
+
+   >>> import dask.dataframe as dd
+
+   >>> df = dd.read_csv('myfile.*.csv')
+   >>> df = df + 100
+   >>> df = df[df.name == 'Alice']
+
+   >>> graph = df.__dask_graph__()
+   >>> graph.layers
+   {
+    'read-csv': {('read-csv', 0): (pandas.read_csv, 'myfile.0.csv'),
+                 ('read-csv', 1): (pandas.read_csv, 'myfile.1.csv'),
+                 ('read-csv', 2): (pandas.read_csv, 'myfile.2.csv'),
+                 ('read-csv', 3): (pandas.read_csv, 'myfile.3.csv')},
+
+    'add': {('add', 0): (operator.add, 'myfile.0.csv', 100),
+            ('add', 1): (operator.add, 'myfile.1.csv', 100),
+            ('add', 2): (operator.add, 'myfile.2.csv', 100),
+            ('add', 3): (operator.add, 'myfile.3.csv', 100)}
```

Comment:
As in the `HighLevelGraph` docstring, guessing `'myfile.*.csv'` should be `('read-csv', *)`.